### PR TITLE
fixes ZEN-20808: Fix for ZEN-17576 needs to be made configurable in d…

### PR DESCRIPTION
**Pull request by HIMix**
…aemon config files.

### CODE REVIEW

File: Products/ZenHub/PBDaemon.py [hunk 1]
```diff
@@ -1100,6 +1100,12 @@ def buildOptions(self):
                                type='int',
                                help='How often to ping zenhub')
 
+        self.parser.add_option('--ping-perspective',
+                               dest='pingPerspective',
+                               help="Enable or disable ping perspective",
+                               default=True,
+                               action='store_true')
```
Comment (jpeacock-zenoss) [inline Products/ZenHub/PBDaemon.py]: This option configuration fails with an error if I try to specify `--ping-perspective=False` on the command line.

**Comment by jplouis** [comment]
This will need a migrate for all collector daemons if you change the option.

**Comment by jplouis** [comment]
There needs to be a migration script so that existing/migrated installs will have the correct options in the conf file.

**Comment by alapidas** [comment]
The migrate script in this PR is incorrect.  It will change config files in the container.  The config files are actually defined in the service definition and then injected into each service.  You need to use the service migration SDK to make these changes on the service definition itself, not in the container.  There are examples that can be followed in the `ZenModel/migrate` folder.

**Pull request closed without merging**